addons/binding/org.openhab.binding.co2signal/src/main/java/org/openhab/binding/co2signal/handler/CO2SignalHandler.java

```diff
@@ -0,0 +1,301 @@
+            }
+        }
+    }
+
+    /**
+     * Get new data from co2cn.org service
+     *
+     * @return {CO2SignalJsonResponse}
+     */
+    private CO2SignalJsonResponse updateCO2SignalData() {
+        CO2SignalConfiguration config = getConfigAs(CO2SignalConfiguration.class);
+        return getCO2SignalData(config);
+    }
+
+    /**
+     * Request new CO2 Signal data to the co2signal.com service
+     *
+     * @param location geo-coordinates from config
+     * @param countryCode station ID from config
+     *
+     * @return the CO2 Signal data object mapping the JSON response or null in case of error
+     */
+    private CO2SignalJsonResponse getCO2SignalData(CO2SignalConfiguration config) {
+        CO2SignalJsonResponse result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        try {
+            String urlStr = buildCO2SignalUrl(config);
+
+            logger.debug("URL = {}", urlStr);
+
+            Properties httpHeaders = new Properties();
+            httpHeaders.put("User-Agent", "Mozilla/5.0");
+            httpHeaders.put("auth-token", config.apikey);
+
+            String response = HttpUtil.executeUrl("GET", urlStr, httpHeaders, null, null, 5000);
+            logger.debug("co2Response = {}", response);
+
+            // Map the JSON response to an object
+            result = gson.fromJson(response, CO2SignalJsonResponse.class);
+
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else if (result.getData() != null && result.getStatus() != "error") {
+                resultOk = true;
+            } else {
+                errorMsg = "missing data sub-object";
+            }
+
+            if (!resultOk) {
+                logger.warn("Error in co2cn.org (CO2 Signal) response: {}", errorMsg);
+            }
+        } catch (MalformedURLException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Constructed url is not valid: {}", errorMsg);
+        } catch (JsonSyntaxException e) {
+            errorMsg = "Configuration is incorrect";
+            logger.warn("Error running co2cn.org (CO2 Signal) request: {}", errorMsg);
+        } catch (IOException | IllegalStateException e) {
+            errorMsg = e.getMessage();
+        }
+
+        // Update the thing status
+        if (resultOk) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, "OK");
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, errorMsg);
+        }
+
+        return resultOk ? result : null;
+    }
+
+    private String buildCO2SignalUrl(CO2SignalConfiguration config) {
+        UriBuilder builder = UriBuilder.fromPath(URL);
+
+        String location = StringUtils.trimToEmpty(config.location);
+        String countryCode = StringUtils.trimToEmpty(config.countryCode);
+
+        if (location != null && !"".equals(location)) {
```

Comment:
You use StringUtils.trimToEmpty() just above, therefore the check against null is not necessary.

In any case, better use StringUtils.isEmpty() here, then even the previous trimToEmpty() is not strictly necessary

Reply:
Modified the code using trimToNull and only check for location  != null. StringUtils.isEmpty() does not catch the case where somebody writes something like that: location="       ", countryCode="    de "  in the config, for a string with only whitespaces isEmpty returns false.